sql/core/src/main/scala/org/apache/spark/sql/execution/stat/FrequentItems.scala

```diff
@@ -66,6 +68,19 @@ object FrequentItems extends Logging {
     }
   }
 
+  /** Helper function to resolve column to expr (if not yet) */
+  // TODO: it might be helpful to have this helper in Dataset.scala,
+  // e.g. `drop` function uses exactly the same flow to deal with
+  // `Column` arguments
+  private def resolveColumn(df: DataFrame, col: Column): Column = {
+    col match {
+      case Column(u: UnresolvedAttribute) =>
+        Column(df.queryExecution.analyzed.resolveQuoted(
+          u.name, df.sparkSession.sessionState.analyzer.resolver).getOrElse(u))
+      case Column(_expr: Expression) => col
+    }
+  }
```

Comment:
The problem with Column is, it can contain an unresolved expression, for example `UnresolvedAttribute + UnresolvedAttribute ...`.

When we are allowed to use column name only, we can rely on `df.resolve(colName)` to resolve it. Once you extend to Column, you cannot do the same check as before.

Reply:
The code here tries to resolve the column if it has `UnresolvedAttribute`. If it still does not provide clarity, I think it's fair to throw an exception. Similar to how `Dataset.drop` works if the argument given is a column with an unresolved attribute.

Reply:
No, I mean for now you only handle the case that `Column(UnresolvedAttribute)`, but `Column` can contain any unresolved expression that can involve many `UnresolvedAttribute`. For the latter one, the added `resolveColumn` cannot resolve it correctly.

Reply:
Exactly the same approach is used for `drop` implementation here: https://github.com/apache/spark/blob/22bb6b0fddb3ecd3ac0ad2b41a5024c86b8a6fc7/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L2500-L2505 -- is there a better approach?

Reply:
As far as we will do `df.select` to execution functions, we can do rely on 

```
df.select(cols: _*).queryExecution.analyzed.output
```

to check data types. `select` will throw exception in case of unresolvable column. @viirya WDYT?